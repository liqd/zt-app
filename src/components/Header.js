import React from 'react'
import { useTranslation } from 'react-i18next'
import { View } from 'react-native'
import IconSLI from 'react-native-vector-icons/SimpleLineIcons'
import { Button } from '@rneui/base'

import { styles } from './Header.styles.js'

export const Header = (props) => {
  const { t } = useTranslation()
  const arrowLeftIcon = (
    <IconSLI
      name='arrow-left'
      size={22}
      color={props.arrowColor
        ? props.arrowColor
        : undefined}
    />
  )
  const backButtonStyle = props.backButtonStyle
    ? props.backButtonStyle
    : styles.backButton
  const backButtonTextStyle = props.backButtonTextStyle
    ? props.backButtonTextStyle
    : styles.backButtonText
  const containerStyle = props.transparent
    ? {}
    : styles.container

  function handleBack() {
    if (props.isEditing !== undefined && props.isEditing) {
      props.toggleEditing()
    } else return props.navigation.goBack()
  }

  const defaultLeftButton = (
    <Button
      buttonStyle={backButtonStyle}
      titleStyle={backButtonTextStyle}
      title={t('Back')}
      type='clear'
      icon={arrowLeftIcon}
      onPress={props.handleCustomBack
        ? props.handleCustomBack
        : handleBack}
    />
  )

  return (
    <View style={containerStyle}>
      <View style={styles.actionsContainer}>
        {props.leftButton
          ? props.leftButton
          : defaultLeftButton}
        {props.rightButton}
      </View>
    </View>
  )
}
