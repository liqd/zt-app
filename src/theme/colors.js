export const COLORS = {
  // brand
  primary: '#2d40cc',
  primaryLight: '#e5e8f8',
  secondary: '#90efe4',
  tertiary: '#008dc5',

  // priority
  success: '#a3ef90',
  info: '#2d40cc',
  warning: '#ffc107',
  danger: '#a72b1e',
  dangerLight: '#f6e9e8',

  // utility
  text: '#1a1a1a',
  paper: '#fff',
  grey: {
    bg: '#f2f2f2',
    extralight: '#d8d8d8',
    light: '#c4c4c4',
    medium: '#666',
    dark: '#464a4c',
    background: '#d8d8d8',
  },
  overlay: 'rgba(77, 77, 77, 0.9)'
}
