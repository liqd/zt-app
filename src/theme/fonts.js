import { PixelRatio } from 'react-native'

const pixelRatio = PixelRatio.getFontScale()

export const FONTS = {
  familySans: 'SourceSansPro_400Regular',
  familySemiBold: 'SourceSansPro_600SemiBold',
}

export const SIZES = {
  xs: pixelRatio * 13,
  sm: pixelRatio * 16,
  base: pixelRatio * 18,
  md: pixelRatio * 20,
  lg: pixelRatio * 22,
  xl: pixelRatio * 28,
  xxl: pixelRatio * 34,
}

export const LINEHEIGHTS = {
  sm: pixelRatio * 21,
  base: pixelRatio * 22,
  md: pixelRatio * 24,
  lg: pixelRatio * 28,
  xl: pixelRatio * 30,
}

export const LETTERSPACING = {
  iconbtn: 5,
}
