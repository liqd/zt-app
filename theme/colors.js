export const COLORS = {
  // brand
  primary: '#2d40cc',
  secondary: '#90efe4',
  tertiary: '#008dc5',

  // priority
  success: '#a3ef90',
  info: '#2d40cc',
  warning: '#ffc107',
  danger: '#a72b1e',

  // utility
  text: {
    main: '#1a1a1a',
  },
  paper: {
    main: '#fff',
  },
  grey: {
    extralight: '#d8d8d8',
    light: '#c4c4c4',
    medium: '#666',
    dark: '#464a4c',
  },
  overlay: 'rgba(77, 77, 77, 0.9)'
}
