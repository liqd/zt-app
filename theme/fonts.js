const baseFontSize = 17;
const baseLineHeight = 22;

export const FONTS = {
  familySans: 'SourceSansPro_400Regular',
};

export const SIZES = {
  xs: baseFontSize * 0.8,
  sm: baseFontSize * 0.85,
  base: baseFontSize,
  md: baseFontSize * 1.2,
  lg: baseFontSize * 1.65,
  xl: baseFontSize * 2,
  multiplyBy: times => times * baseFontSize
};

export const LINEHEIGHTS = {
  sm: baseLineHeight * 0.95,
  base: baseLineHeight,
  md: baseLineHeight * 1.1,
  lg: baseLineHeight * 1.3,
  xl: baseLineHeight * 1.9,
  multiplyBy: times => times * baseLineHeight
};

export const FONTWEIGHT = {
  semiBold: '500',
  bold: '600',
  boldxl: '800',
  boldxxl: 'bold'
};

export const LETTERSPACING = {
  iconbtn: 5,
};
