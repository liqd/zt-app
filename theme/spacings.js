const baseGutter = 20;

export const SPACINGS = {
  base: baseGutter,
  multiplyBy: times => times * baseGutter
};

export const BORDERWIDTH = {
  base: 1,
};

export const BORDERRADIUS = {
  none: 0,
  base: 15,
  container: 24
};
