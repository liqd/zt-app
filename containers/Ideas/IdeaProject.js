import React, { useState, useEffect } from 'react';
import { View, Text, ImageBackground, ScrollView } from 'react-native';
import { Button } from 'react-native-elements';
import { styles } from './IdeaProject.styles';
import { IdeasList } from './IdeasList';
import IconSLI from 'react-native-vector-icons/SimpleLineIcons';
import IconFA from 'react-native-vector-icons/FontAwesome';
import API from '../../BaseApi';
import { COLORS } from '../../theme/colors';
import { DateService } from '../../services/DateService';

export const IdeaProject = (props) => {
  const project = props.navigation.getParam('project');
  const [ideas, setIdeas] = useState([]);
  const [activePhase, setActivePhase] = useState();
  const [phaseStart, setPhaseStart] = useState();
  const [phaseEnd, setPhaseEnd] = useState();
  const bgImage = project.image
    ? project.image
    : 'https://i.imgur.com/o4L0arH.jpg';
  const singleModule = project.single_agenda_setting_module;

  const pressHandler = () =>
    props.navigation.navigate('IdeaCreate', { params: props });

  const plusIcon = <IconSLI name='plus' size={24} color={COLORS.paper.main} />;
  const sortIcon = <IconFA name='filter' size={20} color={COLORS.grey.light} />;
  const filterIcon = (
    <IconFA name='search' size={20} color={COLORS.grey.light} />
  );
  const arrowLeftIcon = (
    <IconSLI name='arrow-left' size={22} color={COLORS.paper.main} />
  );

  useEffect(() => {
    singleModule &&
      API.getIdeas(singleModule).then((ideaResponse) =>
        setIdeas(ideaResponse)
      ) &&
      API.getModule(singleModule).then((moduleResponse) => {
        const aPhase = moduleResponse.phases.find((phase) => phase.is_active);
        aPhase && setActivePhase(aPhase);
        aPhase && setPhaseStart(DateService(aPhase.start_date, 'month d, y, h:m'));
        aPhase && setPhaseEnd(DateService(aPhase.end_date, 'month d, y, h:m'));
      });
  }, []);

  return (
    <View style={styles.container}>
      <ScrollView>
        <ImageBackground style={styles.bgImage} source={{ uri: bgImage }}>
          <View style={styles.actionsContainer}>
            <Button
              buttonStyle={styles.button}
              titleStyle={styles.buttonText}
              title='Back'
              type='clear'
              icon={arrowLeftIcon}
              onPress={() => props.navigation.goBack()}
            />
            <Button
              buttonStyle={styles.button}
              titleStyle={styles.buttonText}
              title='Follow'
              icon={plusIcon}
              type='clear'
            />
          </View>
          <View style={styles.overlayContainer}>
            <View style={styles.infoContainer}>
              <Text style={styles.title}>{project.name}</Text>
              <Text style={styles.description}>{project.description}</Text>
              <Text style={styles.organisation}>By {project.organisation}</Text>
              <View>
                <View style={styles.tabsMenu}>
                  <Text style={styles.tabsMenuItemActive}>Participation</Text>
                  <Text style={styles.tabsMenuItem}>Information</Text>
                  <Text style={styles.tabsMenuItem}>Results</Text>
                </View>
                {activePhase ? (
                  <View style={styles.phaseContainer}>
                    <Text style={styles.phaseText}>
                      {activePhase && activePhase.name + ' (active)'}
                    </Text>
                    <Text style={styles.phaseDate}>
                      {phaseStart} â€“ {phaseEnd}
                    </Text>
                    <Text style={styles.phaseText}>
                      {activePhase && activePhase.description}
                    </Text>
                  </View>
                ) : (
                  <View style={styles.phaseContainer}>
                    <Text>No active phase found.</Text>
                  </View>
                )}
              </View>
            </View>
            {singleModule ? (
              <View>
                <View style={styles.listActions}>
                  <Button icon={filterIcon} type='clear' />
                  <Button icon={sortIcon} type='clear' />
                </View>
                <View style={styles.listContainer}>
                  <IdeasList ideas={ideas} {...props} />
                </View>
                {activePhase && (
                  <Button
                    buttonStyle={styles.submitButton}
                    title='Submit Idea'
                    onPress={pressHandler}
                  />
                )}
              </View>
            ) : (
              <Text>
                Oops. Multiple module projects are not supported in adhocracy+
                app, yet! Please find them on adhocracy+ web.
              </Text>
            )}
          </View>
        </ImageBackground>
      </ScrollView>
    </View>
  );
};
