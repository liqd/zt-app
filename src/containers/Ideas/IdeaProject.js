import React, { useEffect,useState } from 'react'
import { useTranslation } from 'react-i18next'
import { Image,ImageBackground, Linking, ScrollView, View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'
import IconFA from 'react-native-vector-icons/FontAwesome'
import IconSLI from 'react-native-vector-icons/SimpleLineIcons'
import { Button } from '@rneui/themed'
import { LinearGradient } from 'expo-linear-gradient'

import API from '../../BaseApi'
import { ButtonSubmit } from '../../components/ButtonSubmit'
import { Header } from '../../components/Header'
import { LinkTextSourceSans } from '../../components/LinkTextSourceSans'
import { Richtext } from '../../components/Richtext'
import { StatusBarStyled } from '../../components/StatusBarStyled'
import { StickyContainer } from '../../components/StickyContainer'
import { TextSourceSans } from '../../components/TextSourceSans'
import { COLORS } from '../../theme/colors'

import { styles } from './IdeaProject.styles'
import { IdeasList } from './IdeasList'
import { Phase } from './Phase'

export const IdeaProject = (props) => {
  const { t } = useTranslation()
  const tabs = {
    participation: 0,
    information: 1,
    results: 2
  }
  const { project } = props.route.params
  const [ideas, setIdeas] = useState([])
  const [module, setModule] = useState([])
  const [visibleTab, setVisibleTab] = useState(tabs.participation)
  const bgImage = project.image ? project.image : null

  const pressHandler = () =>
    props.navigation.navigate('IdeaCreate', {
      module: module
    })

  const plusIcon = <IconSLI name='plus' size={24} color={COLORS.paper} />
  const sortIcon = <IconFA name='filter' size={20} color={COLORS.grey.light} />
  const filterIcon = (
    <IconFA name='search' size={20} color={COLORS.grey.light} />
  )

  const followButton = (
    <Button
      buttonStyle={styles.button}
      titleStyle={styles.buttonText}
      title={t('Follow')}
      icon={plusIcon}
      type='clear'
      accessible={false}
    />
  )

  const fetchIdeas = () => {
    project.single_idea_collection_module &&
      API.getIdeas(project.single_idea_collection_module)
        .then(ideaResponse => {
          setIdeas(ideaResponse.data)
        })
  }

  const fetchModule = () => {
    API.getModule(project.single_idea_collection_module)
      .then(moduleResponse => {
        setModule(moduleResponse.data)
      })
  }

  useEffect(() => {
    const ideasListener = props.navigation.addListener('focus', () => {
      fetchIdeas()
    })
    return ideasListener
  }, [ideas])

  useEffect(() => {
    fetchModule()
  }, [])

  return (
    <SafeAreaView style={styles.container}>
      <StatusBarStyled
        lightContent={true}
      />
      <ImageBackground
        style={styles.bgImage}
        source={{ uri: bgImage }}
        accessibilityIgnoresInvertColors={true}
      />
      <LinearGradient
        colors={['rgba(0,0,0,0.80)', 'rgba(0,0,0,0.00)']}
        style={styles.linearGradient}
      />
      <ScrollView
        style={visibleTab === tabs.participation && module.has_idea_adding_permission &&
          styles.scrollContainer
        }
      >
        <Header
          transparent={true}
          arrowColor={COLORS.paper}
          backButtonStyle={styles.button}
          backButtonTextStyle={styles.buttonText}
          rightButton={followButton}
          navigation={props.navigation} />
        <View style={styles.overlayContainer}>
          <View style={styles.infoContainer}>
            <TextSourceSans style={styles.title} accessibilityRole="header">
              {project.name}
            </TextSourceSans>
            <TextSourceSans style={styles.description}>
              {project.description}
            </TextSourceSans>
            <View style={styles.organisationContainer}>
              {project.organisation_logo && (
                <View style={styles.organisationLogoContainer}>
                  <Image
                    style={styles.organisationLogo}
                    source={{ uri: project.organisation_logo }}
                    accessibilityIgnoresInvertColors={true}
                  />
                </View>)}
              <TextSourceSans style={styles.organisationName} numberOfLines={1}>
                {project.organisation}
              </TextSourceSans>
            </View>
            <View style={styles.tabsMenu}>
              <Button
                buttonStyle={styles.tabButton}
                titleStyle={visibleTab === tabs.participation ?
                  [ styles.tabsMenuItem, styles.tabsMenuItemActive ] :
                  styles.tabsMenuItem
                }
                title={t('Participation')}
                type='clear'
                onPress={() => setVisibleTab(tabs.participation)}
              />
              <Button
                buttonStyle={styles.tabButton}
                titleStyle={visibleTab === tabs.information ?
                  [ styles.tabsMenuItem, styles.tabsMenuItemActive ] :
                  styles.tabsMenuItem
                }
                title={t('Information')}
                type='clear'
                onPress={() => setVisibleTab(tabs.information)}
              />
              <Button
                buttonStyle={styles.tabButton}
                titleStyle={visibleTab === tabs.results ?
                  [ styles.tabsMenuItem, styles.tabsMenuItemActive ] :
                  styles.tabsMenuItem
                }
                title={t('Results')}
                type='clear'
                onPress={() => setVisibleTab(tabs.results)}
              />
            </View>
            {visibleTab === tabs.participation &&
              <View>
                <Phase
                  activePhase={module.active_phase}
                  futurePhases={module.future_phases}
                  pastPhases={module.past_phases}
                />
                {project.single_idea_collection_module ? (
                  <View style={styles.containerInner}>
                    <View style={styles.listActions}>
                      <Button icon={filterIcon} type='clear' disabled />
                      <Button icon={sortIcon} type='clear' disabled />
                    </View>
                    <View style={styles.listContainer}>
                      <IdeasList
                        ideas={ideas}
                        module={module}
                        navigation={props.navigation}
                      />
                    </View>
                  </View>
                ) : (
                  <TextSourceSans>
                    {t('Oops. Multiple module projects are not supported in adhocracy+ app,'
                        + ' yet! Please find them on adhocracy+ web.')}
                  </TextSourceSans>
                )}
              </View>
            }
            {visibleTab === tabs.information &&
              <View>
                <Richtext text={project.information} />
                {project.has_contact_info &&
                <>
                  <TextSourceSans style={styles.contactHeadline}>
                    {t('Contact for Questions')}
                  </TextSourceSans>
                  {!!project.contact_name &&
                    <TextSourceSans style={styles.contactField}>
                      {project.contact_name}
                    </TextSourceSans>
                  }
                  {!!project.contact_address_text &&
                    <TextSourceSans style={styles.contactField}>
                      {project.contact_address_text}
                    </TextSourceSans>
                  }
                  {!!project.contact_phone &&
                    <TextSourceSans style={styles.contactField}>
                      <TextSourceSans style={styles.contactLabel}>
                        {t('Telephone:')} </TextSourceSans>
                      {project.contact_phone}
                    </TextSourceSans>
                  }
                  {!!project.contact_email &&
                    <TextSourceSans style={styles.contactField}>
                      <TextSourceSans style={styles.contactLabel}>
                        {t('Email:')} </TextSourceSans>
                      <LinkTextSourceSans
                        onPress={() => Linking.openURL(`mailto:${project.contact_email}`)}>
                        {project.contact_email}
                      </LinkTextSourceSans>
                    </TextSourceSans>
                  }
                  {!!project.contact_url &&
                    <TextSourceSans style={styles.contactField}>
                      <TextSourceSans style={styles.contactLabel}>
                        {t('Website:')} </TextSourceSans>
                      <LinkTextSourceSans
                        onPress={() => Linking.openURL(project.contact_url)}>
                        {project.contact_url}
                      </LinkTextSourceSans>
                    </TextSourceSans>
                  }
                </>
                }
              </View>
            }
            {visibleTab === tabs.results &&
              <Richtext text={project.result} />
            }
          </View>
        </View>
      </ScrollView>
      {visibleTab === tabs.participation && module.has_idea_adding_permission && (
        <StickyContainer>
          <ButtonSubmit
            title={t('Submit Idea')}
            onPress={pressHandler}
          ></ButtonSubmit>
        </StickyContainer>
      )}
    </SafeAreaView>
  )
}
