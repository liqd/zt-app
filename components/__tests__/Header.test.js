import React from 'react'
import { StyleSheet } from 'react-native'
import { fireEvent, render } from '@testing-library/react-native'
import { Button } from '@rneui/themed'
import IconSLI from 'react-native-vector-icons/SimpleLineIcons'
import { Header } from '../Header'
import { COLORS } from '../../theme/colors'
import { SIZES } from '../../theme/fonts'
import { SPACINGS } from '../../theme/spacings'
import { styles } from '../../containers/Ideas/IdeaProject.styles'

test('Snapshot Default Header', () => {
  const { toJSON } = render(<Header />)
  expect(toJSON()).toMatchSnapshot()
})

test('Snapshot Default Header Transparent White Arrow', () => {
  const { toJSON } = render(<Header transparent={true} arrowColor={COLORS.paper} />)
  expect(toJSON()).toMatchSnapshot()
})

test('Snapshot Default Header Custom Style', () => {
  const diffStyles = StyleSheet.create({
    backButton: {
      padding: 5,
      marginVertical: 0,
    },
    backButtonText: {
      fontSize: SIZES.md,
      marginHorizontal: SPACINGS.multiplyBy(.25),
      color: COLORS.text,
    }
  })
  const { toJSON } = render(<Header backButtonStyle={diffStyles.backButton} backButtonTextStyle={diffStyles.backButtonText} />)
  expect(toJSON()).toMatchSnapshot()
})

test('Test Custom Header Follow Button Left', () => {
  const plusIcon = <IconSLI name='plus' size={24} color={COLORS.paper} />
  const followButton = (
    <Button
      buttonStyle={styles.button}
      titleStyle={styles.buttonText}
      title='Follow'
      icon={plusIcon}
      type='clear'
    />
  )
  const { getByText, queryByText } = render(<Header leftButton={followButton} />)
  const fb = getByText(/Follow/)
  expect(fb).toBeTruthy()
  const backButton = queryByText(/Back/)
  expect(backButton).toBeFalsy()
})

test('Test Custom Header Follow Button Right', () => {
  const plusIcon = <IconSLI name='plus' size={24} color={COLORS.paper} />
  const followButton = (
    <Button
      buttonStyle={styles.button}
      titleStyle={styles.buttonText}
      title='Follow'
      icon={plusIcon}
      type='clear'
    />
  )
  const { getByText } = render(<Header rightButton={followButton} />)
  const fb = getByText(/Follow/)
  expect(fb).toBeTruthy()
  const backButton = getByText(/Back/)
  expect(backButton).toBeTruthy()
})

test('Test Header Back Button', async () => {
  const navigation = { goBack: jest.fn()}
  const { findByText } = render(<Header navigation={navigation} />)
  const backButton = await findByText('Back')
  fireEvent.press(backButton)
  expect(navigation.goBack).toHaveBeenCalled()
})

test('Test Header Back Button Editing', async () => {
  const toggleEditing = jest.fn()
  const { findByText } = render(<Header isEditing={true} toggleEditing={toggleEditing} />)
  const backButton = await findByText('Back')
  fireEvent.press(backButton)
  expect(toggleEditing).toHaveBeenCalled()
})

test('Test Custom Header Follow Button Right', async () => {
  const plusIcon = <IconSLI name='plus' size={24} color={COLORS.paper} />
  const handleFollow = jest.fn()
  const followButton = (
    <Button
      buttonStyle={styles.button}
      titleStyle={styles.buttonText}
      title='Follow'
      icon={plusIcon}
      type='clear'
      onPress={handleFollow}
    />
  )
  const { findByText } = render(<Header rightButton={followButton} />)
  const fb = await findByText(/Follow/)
  fireEvent.press(fb)
  expect(handleFollow).toHaveBeenCalled()
})
