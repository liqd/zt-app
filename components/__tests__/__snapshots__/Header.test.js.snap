// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Default Header 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 20,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 2,
          },
          undefined,
          false,
        ]
      }
      testID="RNE_BUTTON_WRAPPER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 2,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 0,
              "padding": 0,
              "paddingHorizontal": 10,
            }
          }
        >
          <Text />
          <Text
            style={
              Object {
                "color": "#1a1a1a",
                "fontSize": 17,
                "marginHorizontal": 5,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Snapshot Default Header Custom Style 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 20,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 2,
          },
          undefined,
          false,
        ]
      }
      testID="RNE_BUTTON_WRAPPER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 2,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 0,
              "padding": 5,
              "paddingHorizontal": 10,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 22,
                },
                undefined,
                Object {
                  "fontFamily": "simple-line-icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "#1a1a1a",
                "fontSize": 17,
                "marginHorizontal": 5,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Snapshot Default Header Transparent White Arrow 1`] = `
<View
  style={
    Object {
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 20,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 2,
          },
          undefined,
          false,
        ]
      }
      testID="RNE_BUTTON_WRAPPER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 2,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 0,
              "padding": 0,
              "paddingHorizontal": 10,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 22,
                },
                undefined,
                Object {
                  "fontFamily": "simple-line-icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "#1a1a1a",
                "fontSize": 17,
                "marginHorizontal": 5,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
